### PR TITLE
Convert rst to md for ModelAdmin

### PR DESCRIPTION
**Pull request by thoang43**
_Please check the following:_

Since this is a conversion from rst to md. I am not sure which tests to be run.
- [ ] Do the tests still pass?[^1]
- [X] Does the code comply with the style guide? Documentation guide.
    - [ ] Run `make lint` from the Wagtail root. 
- [ ] For Python changes: Have you added tests to cover the new/fixed behaviour?
- [ ] For front-end changes: Did you test on all of Wagtail’s supported environments?[^2]
    - [ ] **Please list the exact browser and operating system versions you tested**: MacOS, Chrome
    - [ ] **Please list which assistive technologies [^3] you tested**: 
- [ ] For new features: Has the documentation been updated accordingly?

**Please describe additional details for testing this change**. 

[^1]: [Development Testing](https://docs.wagtail.org/en/latest/contributing/developing.html#testing)
[^2]: [Browser and device support](https://docs.wagtail.org/en/latest/contributing/developing.html#browser-and-device-support)
[^3]: [Accessibility Target](https://docs.wagtail.org/en/latest/contributing/developing.html#accessibility-targets)

### CODE REVIEW

**Comment by vupham04** [comment]
Related #8383

**Comment by lb-** [comment]
Thanks so much @vupham04 I will attempt to get this reviewed soon.

File: docs/reference/contrib/modeladmin/create_edit_delete_views.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
+# Customising `CreateView`, `EditView` and `DeleteView`
+
+
+**NOTE:** `modeladmin` only provides 'create', 'edit' and 'delete' functionality for non page type models (i.e. models that do not extend `wagtailcore.models.Page`). If your model is a 'page type' model, customising any of the following will not have any effect:
```
Comment (lb-) [inline docs/reference/contrib/modeladmin/create_edit_delete_views.md:4]: MIght be nice to put this into an admonition I think.

```
```{note}
...
```
```

File: docs/reference/contrib/modeladmin/primer.md [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+local:
+depth: 1
+---
+```
+
+## Wagtail's `ModelAdmin` class isn't the same as Django's
+
+Wagtail's `ModelAdmin` class is designed to be used in a similar way to Django's class of the same name, and it often uses the same attribute and method names to achieve similar things. However, there are a few key differences:
+
+##### Add & edit forms are still defined by `panels` and `edit_handlers`
```
Comment (lb-) [inline docs/reference/contrib/modeladmin/primer.md:16]:
```suggestion
### Add & edit forms are still defined by `panels` and `edit_handlers`
```

Let's change this to heading level 3 - as it is within a heading level 2.

Comment (lb-) [inline docs/reference/contrib/modeladmin/primer.md:16]:
```suggestion
### Add & edit forms are still defined by `panels` and `edit_handlers`
```

Let's change this to heading level 3 - as it is within a heading level 2.

File: docs/reference/contrib/modeladmin/primer.md [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+
+In Wagtail, controlling which fields appear in add/edit forms for your `Model`, and defining how they are grouped and ordered, is achieved by adding a `panels` attribute, or `edit_handler` to your `Model` class. This remains the same whether your model is a `Page` type, a snippet, or just a standard Django `Model`. Because of this, Wagtail's `ModelAdmin` class is mostly concerned with 'listing' configuration. For example, `list_display`, `list_filter` and `search_fields` attributes are present and support largely the same values as Django's ModelAdmin class, while `fields`, `fieldsets`, `exclude` and other attributes you may be used to using to configure Django's add/edit views, simply aren't supported by Wagtail's version.
+
+##### 'Page type' models need to be treated differently from other models
```
Comment (lb-) [inline docs/reference/contrib/modeladmin/primer.md:20]:
```suggestion
### 'Page type' models need to be treated differently from other models
```

Comment (lb-) [inline docs/reference/contrib/modeladmin/primer.md:20]:
```suggestion
### 'Page type' models need to be treated differently from other models
```

File: docs/reference/contrib/modeladmin/primer.md [hunk 4]
```diff
@@ -0,0 +1,326 @@
...
+for them or not).
+
+In order to deliver a consistent user experience, `modeladmin` simply redirects users to Wagtail's existing page management views wherever possible. You should bear this in mind if you ever find yourself wanting to change what happens when pages of a certain type are added, deleted, published, or have some other action applied to them. Customising the `CreateView` or `EditView` for your page type `Model` (even if just to add an additional stylesheet or JavaScript), simply won't have any effect, as those views are not used.
+
+If you do find yourself needing to customise the add, edit or other behaviour for a page type model, you should take a look at the following part of the documentation: [](admin_hooks).
+
+##### Wagtail's `ModelAdmin` class is 'modular'
```
Comment (lb-) [inline docs/reference/contrib/modeladmin/primer.md:31]:
```suggestion
### Wagtail's `ModelAdmin` class is 'modular'
```

Comment (lb-) [inline docs/reference/contrib/modeladmin/primer.md:31]:
```suggestion
### Wagtail's `ModelAdmin` class is 'modular'
```

File: docs/reference/contrib/modeladmin/primer.md [hunk 5]
```diff
@@ -0,0 +1,326 @@
...
+
+- **URL helpers** - That help with the consistent generation, naming and referencing of urls.
+- **Permission helpers** - That help with ensuring only users with sufficient permissions can perform certain actions, or see options to perform those actions.
+- **Button helpers** - That, with the help of the other two, helps with the generation of buttons for use in a number of places.
+
+The `ModelAdmin` class allows you to define and use your own helper classes by setting values on the following attributes:
+
+(modeladmin_url_helper_class)=
+
+##### `ModelAdmin.url_helper_class`
```
Comment (lb-) [inline docs/reference/contrib/modeladmin/primer.md:168]:
```suggestion
### `ModelAdmin.url_helper_class`
```

Comment (lb-) [inline docs/reference/contrib/modeladmin/primer.md:168]:
```suggestion
### `ModelAdmin.url_helper_class`
```

File: docs/reference/contrib/modeladmin/primer.md [hunk 6]
```diff
@@ -0,0 +1,326 @@
...
+
+        def get_url_helper_class(self):
+            if self.some_attribute is True:
+                return MyURLHelper
+            return AdminURLHelper
+```
+
+(modeladmin_permission_helper_class)=
+
+##### `ModelAdmin.permission_helper_class`
```
Comment (lb-) [inline docs/reference/contrib/modeladmin/primer.md:207]:
```suggestion
### `ModelAdmin.permission_helper_class`
```

Comment (lb-) [inline docs/reference/contrib/modeladmin/primer.md:207]:
```suggestion
### `ModelAdmin.permission_helper_class`
```

File: docs/reference/contrib/modeladmin/primer.md [hunk 7]
```diff
@@ -0,0 +1,326 @@
...
+
+        def get_permission_helper_class(self):
+            if self.some_attribute is True:
+                return MyPermissionHelper
+            return PermissionHelper
+```
+
+(modeladmin_button_helper_class)=
+
+##### `ModelAdmin.button_helper_class`
```
Comment (lb-) [inline docs/reference/contrib/modeladmin/primer.md:244]:
```suggestion
### `ModelAdmin.button_helper_class`
```

Comment (lb-) [inline docs/reference/contrib/modeladmin/primer.md:244]:
```suggestion
### `ModelAdmin.button_helper_class`
```

File: docs/reference/contrib/modeladmin/primer.md [hunk 8]
```diff
@@ -0,0 +1,326 @@
...
+
+        def get_button_helper_class(self):
+            if self.some_attribute is True:
+                return MyButtonHelper
+            return ButtonHelper
+```
+
+(modeladmin_helpers_in_custom_views)=
+
+##### Using helpers in your custom views
```
Comment (lb-) [inline docs/reference/contrib/modeladmin/primer.md:317]:
```suggestion
### Using helpers in your custom views
```

Comment (lb-) [inline docs/reference/contrib/modeladmin/primer.md:317]:
```suggestion
### Using helpers in your custom views
```

File: docs/reference/contrib/modeladmin/index.md [hunk 9]
```diff
@@ -0,0 +1,190 @@
...
+indexview
+create_edit_delete_views
+inspectview
+chooseparentview
+tips_and_tricks/index
+```
+
+(modeladmin_usage)=
+
+#### Installation
```
Comment (lb-) [inline docs/reference/contrib/modeladmin/index.md:38]:
```suggestion
### Installation
```

Comment (lb-) [inline docs/reference/contrib/modeladmin/index.md:38]:
```suggestion
### Installation
```

File: docs/reference/contrib/modeladmin/index.md [hunk 10]
```diff
@@ -0,0 +1,190 @@
...
+Add `wagtail.contrib.modeladmin` to your `INSTALLED_APPS`:
+
+```python
+    INSTALLED_APPS = [
+       ...
+       'wagtail.contrib.modeladmin',
+    ]
+```
+
+#### How to use
```
Comment (lb-) [inline docs/reference/contrib/modeladmin/index.md:49]:
```suggestion
### How to use
```

Comment (lb-) [inline docs/reference/contrib/modeladmin/index.md:49]:
```suggestion
### How to use
```

File: docs/reference/contrib/modeladmin/index.md [hunk 11]
```diff
@@ -0,0 +1,190 @@
...
+
+
+(modeladmin_example_simple)=
+
+##### A simple example
```
Comment (lb-) [inline docs/reference/contrib/modeladmin/index.md:54]:
```suggestion
### A simple example
```

Comment (lb-) [inline docs/reference/contrib/modeladmin/index.md:54]:
```suggestion
### A simple example
```

File: docs/reference/contrib/modeladmin/index.md [hunk 12]
```diff
@@ -0,0 +1,190 @@
...
+        list_filter = ('author',)
+        search_fields = ('title', 'author')
+
+    # Now you just need to register your customised ModelAdmin class with Wagtail
+    modeladmin_register(BookAdmin)
+```
+
+(modeladmin_example_complex)=
+
+##### A more complicated example
```
Comment (lb-) [inline docs/reference/contrib/modeladmin/index.md:112]:
```suggestion
### A more complicated example
```

Comment (lb-) [inline docs/reference/contrib/modeladmin/index.md:112]:
```suggestion
### A more complicated example
```

File: docs/reference/contrib/modeladmin/index.md [hunk 13]
```diff
@@ -0,0 +1,190 @@
...
+        items = (BookAdmin, AuthorAdmin, GenreAdmin)
+
+    # When using a ModelAdminGroup class to group several ModelAdmin classes together,
+    # you only need to register the ModelAdminGroup class with Wagtail:
+    modeladmin_register(LibraryGroup)
+```
+
+(modeladmin_multi_registration)=
+
+##### Registering multiple classes in one `wagtail_hooks.py` file
```
Comment (lb-) [inline docs/reference/contrib/modeladmin/index.md:168]:
```suggestion
### Registering multiple classes in one `wagtail_hooks.py` file
```

Comment (lb-) [inline docs/reference/contrib/modeladmin/index.md:168]:
```suggestion
### Registering multiple classes in one `wagtail_hooks.py` file
```

**Review by lb- — CHANGES_REQUESTED** [review]
@thoang43 thanks for taking the time to prepare this, it looks pretty good.
I have a few recommended fixes but in general - be sure to run `make clean` & `make html` to check for warnings, the main warnings introduced are for the heading level issues.

It looks like some heading levels were not correctly brought over to the MD files.


reference/contrib/modeladmin/index.md:38: WARNING: Non-consecutive header level increase; H2 to H4 [myst.header]
reference/contrib/modeladmin/index.md:49: WARNING: Non-consecutive header level increase; H2 to H4 [myst.header]
reference/contrib/modeladmin/primer.md:16: WARNING: Non-consecutive header level increase; H2 to H5 [myst.header]
reference/contrib/modeladmin/primer.md:20: WARNING: Non-consecutive header level increase; H2 to H5 [myst.header]
reference/contrib/modeladmin/primer.md:31: WARNING: Non-consecutive header level increase; H2 to H5 [myst.header]
reference/contrib/modeladmin/primer.md:168: WARNING: Non-consecutive header level increase; H2 to H5 [myst.header]
reference/contrib/modeladmin/primer.md:207: WARNING: Non-consecutive header level increase; H2 to H5 [myst.header]
reference/contrib/modeladmin/primer.md:244: WARNING: Non-consecutive header level increase; H2 to H5 [myst.header]
reference/contrib/modeladmin/primer.md:317: WARNING: Non-consecutive header level increase; H2 to H5 [myst.header]

Also this error came up 
* reference/contrib/modeladmin/tips_and_tricks/reversing_urls.rst:134: WARNING: Failed to create a cross reference. A title or caption not found: modeladmin_url_helper_class - it could be that `docs/reference/contrib/modeladmin/tips_and_tricks/reversing_urls.rst` is not picking up the reference correctly for some reason.

Who should we put as the contributor on this, yourself or Amelia Dao (or both).
* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *

**Comment by thoang43** [comment]
Hi LB, thank you for all your suggestion. I believe the problem with reversing_urls.rst is that I did not convert that file yet. The jumping heading warning did show when I ran the make. Since I just naively turned all the title level from rst to markdown, the existed jumping heading in rst created these warnings. Do you want it to be the same with the old rst or change the new heading level? I will implement the note admonition as you suggested.

**Comment by vupham04** [comment]
@lb- hi, I have made the changes, but I do want to note that the old rst file has already jumped the heading so we just follow. But I think that this correct-formatted heading helps fix the cross reference error.

About the author, if possible, can you make @thoang43 as the main author, and I'm the co-author (Co-authored-by: Vu Pham <46913707+vupham04@users.noreply.github.com>)? Thank you

**Comment by lb-** [comment]
Thanks @vupham04 - I'll take another look. 
Yeah it's a balance between fixing things while doing these kinds of changes but for something like heading levels it's good to clean it up.

**Review by lb- — APPROVED** [review]
Looking great, I have made some tweaks and converted the last page that was in this folder (reversing URLs).

Thanks so much for this @vupham04  & @thoang43

**Pull request merged**